lib/elixir/lib/macro.ex

```diff
@@ -313,6 +315,21 @@ defmodule Macro do
             "Kernel.#{op}(#{to_string(arg)}), instead of #{op}#{to_string(arg)}"
   end
 
+  def pipe(
+        expr,
+        {{_, meta, [Access, :get] = op}, _line, args} = _op_args,
+        integer
+      )
+      when is_list(args) do
+    if {:from_brackets, true} in meta do
+      raise ArgumentError,
+            "cannot pipe into expression ending in a bracket-based Access.get/3. It is likely you intended to pipe " <>
+              "#{to_string(expr)} into #{to_string(List.first(args))}. Please call Access.get/3 directly rather than using brackets."
```

Comment:
I haven't tested it, but what do you think about this error message:

```suggestion
      raise ArgumentError, """
      wrong operator precedence when piping into bracket-based access
      
      Instead of:

          #{to_string(expr)} |> #{to_string(first)}[#{to_string(second)}]

    You should write:

          (#{to_string(expr)} |> #{to_string(first)})[#{to_string(second)}]
    """
```

Where you will replace `args` by `[first, second]`. WDYT?

Reply:
I definitely think this is a better error message, and is a lot more helpful and direct. I'll swap over to this!